plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -120,27 +126,27 @@ public synchronized Compute client() {
         }
 
         try {
-            HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
-            JSON_FACTORY = new JacksonFactory();
+            gceJsonFactory = new JacksonFactory();
 
             logger.info("starting GCE discovery service");
-
-            ComputeCredential credential = new ComputeCredential.Builder(HTTP_TRANSPORT, JSON_FACTORY)
-                    .setTokenServerEncodedUrl(TOKEN_SERVER_ENCODED_URL)
+            ComputeCredential credential = new ComputeCredential.Builder(getGceHttpTransport(), gceJsonFactory)
+                        .setTokenServerEncodedUrl(TOKEN_SERVER_ENCODED_URL)
                     .build();
 
             credential.refreshToken();
 
             logger.debug("token [{}] will expire in [{}] s", credential.getAccessToken(), credential.getExpiresInSeconds());
-            refreshInterval = TimeValue.timeValueSeconds(credential.getExpiresInSeconds()-1);
+            if (credential.getExpiresInSeconds() != null) {
+                refreshInterval = TimeValue.timeValueSeconds(credential.getExpiresInSeconds()-1);
+            }
 
             // Once done, let's use this token
-            this.client = new Compute.Builder(HTTP_TRANSPORT, JSON_FACTORY, null)
+            this.client = new Compute.Builder(getGceHttpTransport(), gceJsonFactory, null)
                     .setApplicationName(Fields.VERSION)
                     .setHttpRequestInitializer(credential)
                     .build();
         } catch (Exception e) {
-            logger.warn("unable to start GCE discovery service: {} : {}", e.getClass().getName(), e.getMessage());
+            logger.warn("unable to start GCE discovery service: {} : {}", e, e.getClass().getName(), e.getMessage());
```

Comment:
This looks not quite right. I'm not 100% sure, but I'd do:
```
+            logger.warn("unable to start GCE discovery service", e);
```
and let the logger do its job with the exception. I'm tired though - airplanes. So don't take what I type at face value.